src/StreamSelectLoop.php

```diff
@@ -220,27 +213,124 @@ private function waitForStreamActivity($timeout)
     }
 
     /**
+     * Returns integer amount of seconds in $time or null.
+     *
+     * @param float|null $time – time in seconds
+     *
+     * @return int|null
+     */
+    private static function getSeconds($time)
+    {
+        if ($time === null) {
+            return null;
+        }
+
+        /*
+         * Workaround for PHP int overflow:
+         * (float)PHP_INT_MAX == PHP_INT_MAX => true
+         * (int)(float)PHP_INT_MAX == PHP_INT_MAX => false
+         * (int)(float)PHP_INT_MAX == PHP_INT_MIN => true
+         */
+        if ($time == PHP_INT_MAX) {
+            return PHP_INT_MAX;
+        }
+
+        return intval(floor($time));
+    }
+
+    /**
+     * Returns integer amount of microseconds in $time or null.
+     *
+     * @param float|null $time – time in seconds
+     *
+     * @return int|null
+     */
+    private static function getMicroseconds($time)
+    {
+        if ($time === null) {
+            return null;
```

Comment:
Should probably return `0`. `stream_select` accepts `null` only for the `$seconds` parameter.